lib/gemonames.rb

```diff
@@ -7,15 +7,34 @@ module Gemonames
   module_function
   BASE_API_URL = "http://api.geonames.org"
 
-  def client(username:, connection: nil)
-    connection ||= build_connection(username: username)
+  def client(username: nil, connection: nil, token: nil)
```

Comment:
I don't know TBH. I don't like it either. We want other people to use this gem. I don't think dropping the `username` support is good.

Would `client_by_username` and `client_by_token` make you happy?